apps/homescreen/js/pages.js

```diff
@@ -219,9 +219,10 @@
       // Remove child immediately, datastore operations can be quite slow
       this.removeCard(this.selectedCard);
 
+      var revId = this.pagesStore.datastore.revisionId;
       this.pagesStore.get(id).then(entry => {
         entry.data.pinned = false;
-        this.pagesStore.datastore.put(entry.data, id).then(() => {},
+        this.pagesStore.datastore.put(entry.data, id, revId).then(() => {},
```

Comment:
It's incredibly unlikely we hit a race here, right? Just wondering if we want to have some kind of retry code, or if it's just fine for us to fail. Note that we've removed the card from the container, so when the homescreen restarts, the pinned page will reappear. If it's unlikely enough that we can expect it never to happen, I'm happy to not do that.